org.eclipse.scanning.api/src/org/eclipse/scanning/api/points/AbstractGenerator.java

```diff
@@ -376,7 +375,8 @@ public boolean equals(Object obj) {
 		} else if (!description.equals(other.description))
 			return false;
 		if (enabled != other.enabled)
-			return false;
+			return false;if (containers.size()<1) return true;
+
 		if (iconPath == null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This line will not be executed conditionally; only the first line of this 3-line block will be. The rest will execute unconditionally. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2681)